src/handlers/actions/moveFunds.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  AnyColonyClient,
+  ColonyClientV1,
+  ColonyClientV2,
+  ColonyClientV3,
+  ColonyClientV4,
+} from '@colony/colony-js';
+import { BigNumber } from 'ethers';
+import networkClient from '~/networkClient';
+import { ColonyActionType, ContractEvent } from '~/types';
+import { toNumber, writeActionFromEvent } from '~/utils';
+import { getDatabaseDomainId } from '~/utils/domains';
+
+/**
+ * The handler makes use of colonyClient getDomainFromFundingPot method which is only
+ * available on ColonyClientV5 and above. The following type predicate allows to check
+ * we're dealing with a client that supports this method
+ */
+type SupportedColonyClient = Exclude<
+  AnyColonyClient,
+  ColonyClientV1 | ColonyClientV2 | ColonyClientV3 | ColonyClientV4
+>;
+const isSupportedColonyClient = (
+  colonyClient: AnyColonyClient,
+): colonyClient is SupportedColonyClient =>
+  (colonyClient as SupportedColonyClient).getDomainFromFundingPot !== undefined;
+
+export default async (event: ContractEvent): Promise<void> => {
+  const { contractAddress: colonyAddress } = event;
+  const {
+    agent: initiatorAddress,
+    token: tokenAddress,
+    amount,
+    fromPot,
+    toPot,
+  } = event.args;
+
+  const colonyClient = await networkClient.getColonyClient(colonyAddress);
```

Comment:
I wonder if we could optimize the ingestor somewhat if we were to store all known instances of the colony client.

I see us re-instantiating the client for a lot of tracked events.

But I actually don't know what this impact would be real-life.